### PR TITLE
Check product blueprints setting to show/hide Blueprints accordion

### PR DESCRIPTION
**Pull request by h-kataria**
- Added REST API call to check whether to show/hide blueprints accordion
- Until product features for blueprints are added, check if user has access to view or add/edit catalog item features for blueprints accordion.

### CODE REVIEW

**Comment by h-kataria** [comment]
@AparnaKarve please review/test

File: client/app/services/session.service.js [hunk 1]
```diff
@@ -87,7 +94,7 @@
         services:    {show: entitledForServices(productFeatures)},
         requests:    {show: entitledForRequests(productFeatures)},
         marketplace: {show: entitledForServiceCatalogs(productFeatures)},
-        blueprints:  {show: false}
+        blueprints:  {show: entitledForServiceCatalogs(productFeatures) && $state["blueprints_flag"]}
```
Comment (dclarizio) [inline client/app/services/session.service.js]: @h-kataria this should be checking the feature that give the Catalog Items accordion in the OPS UI

File: client/app/services/session.service.js [hunk 2]
```diff
@@ -43,7 +43,14 @@
       delete $sessionStorage.token;
     }
 
+    function fetch_product_setting(key_name, key){
+      $http.get('/api/settings').then(function(response) {
```
Comment (AparnaKarve) [inline client/app/services/session.service.js]: Lets go with the camelCase style for naming JS functions, so that would be ```fetchProductSetting```

**Comment by dclarizio** [comment]
Added WIP until https://github.com/ManageIQ/manageiq/pull/7700 is merged.

**Comment by h-kataria** [comment]
@dclarizio @AparnaKarve addressed your comments

**Comment by h-kataria** [comment]
@dclarizio removing WIP as https://github.com/ManageIQ/manageiq/pull/7700 is merged.

@AparnaKarve please review/test

File: client/app/services/session.service.js [hunk 3]
```diff
@@ -117,6 +124,26 @@
       return angular.isDefined(serviceFeature);
     }
 
+    function isAnyActionAllowed(actions, productFeatures) {
+      for (var i = 0; i <= actions.length; i++) {
+        if (angular.isDefined(actions[i])) {
```
Comment (AparnaKarve) [inline client/app/services/session.service.js]: @h-kataria This method does not seem to work due to the reasons below -
1. `actions[i]` is merely a string here and `angular.isDefined` would always return a `true` for that
2. What `angular.isDefined` needs to look at is the object `productFeatures.atomic_catalogitem_edit`

So this is what I recommend -
1. Let `actions` be an array of the feature names
```javascript
var actions = ["atomic_catalogitem_edit",
               "atomic_catalogitem_new",
               "catalog_items_view",
               "catalogitem_edit",
               "catalogitem_new"];
```
2. Make sure you reference these feature names via the `productFeatures` object
```javascript
if (angular.isDefined(productFeatures[actions[i]]))
```

**Comment by h-kataria** [comment]
@AparnaKarve made suggested changes, please re-review. Thx for the feedback.

**Comment by AparnaKarve** [comment]
@h-kataria Looks good now :+1: 

OK for a merge.

**Comment by h-kataria** [comment]
@dclarizio @martinpovolny please merge

**Pull request merged**